src/pudl_archiver/orchestrator.py

```diff
@@ -188,14 +188,15 @@ async def run(self) -> RunSummary:
 
         draft = await self.depositor.get_deposition_by_id(draft.id_)
         new_datapackage, old_datapackage = await self._update_datapackage(
-            draft, resources
+            original=original, draft=draft, resources=resources
         )
         run_summary = self._summarize_run(
             old_datapackage, new_datapackage, resources, draft.links.html
         )
 
-        if not self.changes:
-            # must run *after* potentially updating datapackage.json
+        if len(run_summary.file_changes) == 0 and not self._datapackage_worth_changing(
```

Comment:
Still don't love this `_datapackage_worth_changing` thing.

Reply:
We call `_summarize_run` before this, and it takes both datapackages, so maybe we could `run_summary` contain a field that tells us if files changed, or the datapackage changed?

Reply:
Yeah, that could work! I think this is probably OK for now and then we can fix it later if we revisit the code 😬